From 862fee60a2f7a7dce54eb10da1d1e8025c9e7dc7 Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Fri, 12 Nov 2021 16:53:43 -0800
Subject: [PATCH 4/5] No lodepng

---
 third_party/CMakeLists.txt | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/third_party/CMakeLists.txt b/third_party/CMakeLists.txt
index 64e6035..d1fc04a 100644
--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -103,13 +103,7 @@ else()
 endif()
 
 # lodepng
-if( NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/lodepng/lodepng.h" )
-  message(FATAL_ERROR "Please run ${PROJECT_SOURCE_DIR}/deps.sh to fetch the "
-          "build dependencies.")
-endif()
-include(lodepng.cmake)
-configure_file("${CMAKE_CURRENT_SOURCE_DIR}/lodepng/LICENSE"
-               ${PROJECT_BINARY_DIR}/LICENSE.lodepng COPYONLY)
+
 
 # brotli
 if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/brotli/c/include/brotli/decode.h" OR
-- 
2.34.0.windows.1

