From 304ca775222f022699c91a782eb777604106924a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sat, 18 Sep 2021 12:24:45 +0200
Subject: [PATCH] Increase label size from 48 -> 128

---
 src/psl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/psl.c b/src/psl.c
index f1691e0..617753d 100644
--- a/src/psl.c
+++ b/src/psl.c
@@ -129,7 +129,7 @@ typedef SSIZE_T ssize_t;
 
 typedef struct {
 	char
-		label_buf[48];
+		label_buf[128];
 	const char *
 		label;
 	unsigned short
