diff --git a/ssl/tls13_lib.c b/ssl/tls13_lib.c
index 9dbb7d6430..f064521c8b 100644
--- a/ssl/tls13_lib.c
+++ b/ssl/tls13_lib.c
@@ -147,7 +147,8 @@ tls13_alert_sent_cb(uint8_t alert_desc, void *arg)
 	}
 
 	/* All other alerts are treated as fatal in TLSv1.3. */
-	SSLerror(ctx->ssl, SSL_AD_REASON_OFFSET + alert_desc);
+	if (ctx->error.code == 0)
+		SSLerror(ctx->ssl, SSL_AD_REASON_OFFSET + alert_desc);
 }
 
 static void
