From 15317d058691d5739bd55da77ceb5c23587286e2 Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Mon, 26 Nov 2018 16:32:05 -0800
Subject: [PATCH] Adjust CMake for vcpkg

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6957f619d..67479efef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1359,7 +1359,7 @@ endfunction()
 
 include(GNUInstallDirs)
 
-set(CURL_INSTALL_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+set(CURL_INSTALL_CMAKE_DIR ${CMAKE_INSTALL_DATADIR}/curl/cmake)
 set(TARGETS_EXPORT_NAME "${PROJECT_NAME}Targets")
 set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
 set(project_config "${generated_dir}/${PROJECT_NAME}Config.cmake")
-- 
2.36.1.windows.1

