From 1faa8c6b2fcd52b314010fd80cdd5f49f0154ffd Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Fri, 12 Nov 2021 12:18:24 -0800
Subject: [PATCH 4/4] Make additional libraries optional

---
 CMakeLists.txt | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9058f48..3ad0843 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,6 +46,9 @@ set(HWY_CMAKE_ARM7 OFF CACHE BOOL "Set copts for ARMv7 with NEON?")
 # arise due to compiler/platform changes. Enable this in CI/tests.
 set(HWY_WARNINGS_ARE_ERRORS OFF CACHE BOOL "Add -Werror flag?")
 
+set(HWY_CONTRIB ON CACHE BOOL "Build hwy_contrib")
+set(HWY_TEST ON CACHE BOOL "Build hwy_test")
+
 set(HWY_ENABLE_EXAMPLES ON CACHE BOOL "Build examples")
 set(HWY_ENABLE_INSTALL ON CACHE BOOL "Install library")
 
@@ -198,15 +201,19 @@ target_compile_options(hwy PRIVATE ${HWY_FLAGS})
 set_property(TARGET hwy PROPERTY POSITION_INDEPENDENT_CODE ON)
 target_include_directories(hwy PUBLIC ${CMAKE_CURRENT_LIST_DIR})
 
+if (HWY_CONTRIB)
 add_library(hwy_contrib STATIC ${HWY_CONTRIB_SOURCES})
 target_compile_options(hwy_contrib PRIVATE ${HWY_FLAGS})
 set_property(TARGET hwy_contrib PROPERTY POSITION_INDEPENDENT_CODE ON)
 target_include_directories(hwy_contrib PUBLIC ${CMAKE_CURRENT_LIST_DIR})
+endif()
 
+if (HWY_TEST)
 add_library(hwy_test STATIC ${HWY_TEST_SOURCES})
 target_compile_options(hwy_test PRIVATE ${HWY_FLAGS})
 set_property(TARGET hwy_test PROPERTY POSITION_INDEPENDENT_CODE ON)
 target_include_directories(hwy_test PUBLIC ${CMAKE_CURRENT_LIST_DIR})
+endif()
 
 # -------------------------------------------------------- hwy_list_targets
 # Generate a tool to print the compiled-in targets as defined by the current
@@ -244,6 +251,9 @@ foreach (source ${HWY_SOURCES})
   endif()
 endforeach()
 
+set(HWY_PC libhwy.pc)
+
+if (HWY_CONTRIB)
 install(TARGETS hwy_contrib
   DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 # Install all the headers keeping the relative path to the current directory
@@ -255,7 +265,10 @@ foreach (source ${HWY_CONTRIB_SOURCES})
         DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${dirname}")
   endif()
 endforeach()
+list(APPEND HWY_PC libhwy-contrib.pc)
+endif()
 
+if (HWY_TEST)
 install(TARGETS hwy_test
   DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 # Install all the headers keeping the relative path to the current directory
@@ -267,10 +280,12 @@ foreach (source ${HWY_TEST_SOURCES})
         DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${dirname}")
   endif()
 endforeach()
+list(APPEND HWY_PC libhwy-test.pc)
+endif()
 
 # Add a pkg-config file for libhwy and the contrib/test libraries.
 set(HWY_LIBRARY_VERSION "${CMAKE_PROJECT_VERSION}")
-foreach (pc libhwy.pc libhwy-contrib.pc libhwy-test.pc)
+foreach (pc ${HWY_PC})
   configure_file("${CMAKE_CURRENT_SOURCE_DIR}/${pc}.in" "${pc}" @ONLY)
   install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${pc}"
       DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
-- 
2.33.0.windows.2

