From 6d9429ccdc5e794f1637e272fed2985b1fffa83d Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Mon, 29 Aug 2022 15:23:25 -0700
Subject: [PATCH 4/6] Adjust CMake for vcpkg

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1567cfbd1..031b298d8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -674,6 +674,6 @@ install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 install(EXPORT OpenSSLTargets
   FILE OpenSSLTargets.cmake
   NAMESPACE OpenSSL::
-  DESTINATION lib/cmake/OpenSSL
+  DESTINATION ${CMAKE_INSTALL_DATADIR}/boringssl/cmake
 )
-install(FILES OpenSSLConfig.cmake DESTINATION lib/cmake/OpenSSL)
+install(FILES OpenSSLConfig.cmake DESTINATION ${CMAKE_INSTALL_DATADIR}/boringssl/cmake)
-- 
2.37.2.windows.2

