From b58a2bbd412274ec39248609b27c5923f41518a6 Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Mon, 28 Feb 2022 10:53:57 -0800
Subject: [PATCH 5/5] Add install targets

---
 CMakeLists.txt | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cb3b390fb..172d8aa28 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -660,3 +660,16 @@ add_custom_target(
     DEPENDS all_tests bssl_shim handshaker fips_specific_tests_if_any
     USES_TERMINAL)
 endif()
+
+include(GNUInstallDirs)
+
+install(TARGETS crypto ssl
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
+install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+if(ENABLE_TOOLS)
+  install(TARGETS bssl DESTINATION tools/boringssl)
+endif()
-- 
2.37.0.windows.1

