From 76a5f4cd37ec852bbc227e689e35c98d3d8c9db0 Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Thu, 6 Dec 2018 18:03:08 -0800
Subject: [PATCH 1/2] Adjust CMake for vcpkg

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 019e205b..cc2e1623 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -795,7 +795,7 @@ if (NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
   )
   install(EXPORT harfbuzzConfig
       NAMESPACE harfbuzz::
-      DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/harfbuzz
+      DESTINATION ${CMAKE_INSTALL_DATADIR}/harfbuzz/cmake
   )
   if (HB_BUILD_UTILS)
     if (WIN32 AND BUILD_SHARED_LIBS)
-- 
2.19.0.windows.1

