From 651c1712c410585d9f356cae102e296521aa7aa4 Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Tue, 5 Nov 2019 16:56:34 -0800
Subject: [PATCH 1/2] Use find_package for ICU

---
 CMakeLists.txt | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9a1d9032f..5ca958130 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -245,12 +245,7 @@ endif ()
 if (HB_HAVE_ICU)
   add_definitions(-DHAVE_ICU)
 
-  # https://github.com/WebKit/webkit/blob/fdd7733f2f30eab7fe096a9791f98c60f62f49c0/Source/cmake/FindICU.cmake
-  find_package(PkgConfig)
-  pkg_check_modules(PC_ICU QUIET icu-uc)
-
-  find_path(ICU_INCLUDE_DIR NAMES unicode/utypes.h HINTS ${PC_ICU_INCLUDE_DIRS} ${PC_ICU_INCLUDEDIR})
-  find_library(ICU_LIBRARY NAMES libicuuc cygicuuc cygicuuc32 icuuc HINTS ${PC_ICU_LIBRARY_DIRS} ${PC_ICU_LIBDIR})
+  find_package(ICU REQUIRED COMPONENTS uc)
 
   include_directories(${ICU_INCLUDE_DIR})
 
-- 
2.36.0.windows.1

