From 9e6310ba679c3afab768cb0e82b2433e9aa2a810 Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Tue, 5 Nov 2019 16:57:32 -0800
Subject: [PATCH 2/2] Adjust CMake for vcpkg

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9813799b5..482d95916 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -706,7 +706,7 @@ if (NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
   )
   install(EXPORT harfbuzzConfig
       NAMESPACE harfbuzz::
-      DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/harfbuzz
+      DESTINATION ${CMAKE_INSTALL_LIBDIR}/harfbuzz/cmake
   )
   if (HB_HAVE_ICU)
     install(TARGETS harfbuzz-icu
-- 
2.37.1.windows.1

