From e97542a70ed23ce935465a051ae3a8c110b46722 Mon Sep 17 00:00:00 2001
From: Don Olmstead <don.j.olmstead@gmail.com>
Date: Fri, 5 Jul 2024 10:50:51 -0700
Subject: [PATCH 1/2] Remove library suffix on Windows

---
 CMakeLists.txt | 38 --------------------------------------
 1 file changed, 38 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fb352475..0df2698e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -247,25 +247,6 @@ set_target_properties(
 	SOVERSION ${LIBXSLT_MAJOR_VERSION}
 )
 
-if(MSVC)
-	if(BUILD_SHARED_LIBS)
-		set_target_properties(
-			LibXslt
-			PROPERTIES
-			DEBUG_POSTFIX d
-		)
-	else()
-		set_target_properties(
-			LibXslt
-			PROPERTIES
-			DEBUG_POSTFIX sd
-			MINSIZEREL_POSTFIX s
-			RELEASE_POSTFIX s
-			RELWITHDEBINFO_POSTFIX s
-		)
-	endif()
-endif()
-
 install(FILES ${LIBXSLT_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libxslt COMPONENT development)
 
 install(
@@ -340,25 +321,6 @@ set_target_properties(
 	SOVERSION ${LIBEXSLT_MAJOR_VERSION}
 )
 
-if(MSVC)
-	if(BUILD_SHARED_LIBS)
-		set_target_properties(
-			LibExslt
-			PROPERTIES
-			DEBUG_POSTFIX d
-		)
-	else()
-		set_target_properties(
-			LibExslt
-			PROPERTIES
-			DEBUG_POSTFIX sd
-			MINSIZEREL_POSTFIX s
-			RELEASE_POSTFIX s
-			RELWITHDEBINFO_POSTFIX s
-		)
-	endif()
-endif()
-
 install(FILES ${LIBEXSLT_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libexslt COMPONENT development)
 
 install(
-- 
2.45.2.windows.1

