From 34ec4d69dbb65bc42ec89cef9ce7960f00a64d0e Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Thu, 13 Jun 2024 15:32:36 -0700
Subject: [PATCH] Remove library suffix on Windows

---
 CMakeLists.txt | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f99fd368..a3cda8dd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -438,25 +438,6 @@ set_target_properties(
     SOVERSION ${LIBXML_MAJOR_VERSION}
 )
 
-if(MSVC)
-    if(BUILD_SHARED_LIBS)
-        set_target_properties(
-            LibXml2
-            PROPERTIES
-            DEBUG_POSTFIX d
-        )
-    else()
-        set_target_properties(
-            LibXml2
-            PROPERTIES
-            DEBUG_POSTFIX sd
-            MINSIZEREL_POSTFIX s
-            RELEASE_POSTFIX s
-            RELWITHDEBINFO_POSTFIX s
-        )
-    endif()
-endif()
-
 install(FILES ${LIBXML2_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libxml2/libxml COMPONENT development)
 
 install(
-- 
2.46.0.windows.1
