From ca8f8749433a6eaf319de44826761a36c3be427c Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Tue, 23 Aug 2022 14:49:44 -0700
Subject: [PATCH 1/2] Adjust CMake for vcpkg

---
 CMakeLists.txt | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 834b35d3..9d2a8f4f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -584,12 +584,12 @@ install(DIRECTORY doc/ DESTINATION ${CMAKE_INSTALL_DOCDIR} COMPONENT documentati
 
 configure_package_config_file(
 	libxml2-config.cmake.cmake.in libxml2-config.cmake
-	INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libxml2-${PROJECT_VERSION}
+	INSTALL_DESTINATION ${CMAKE_INSTALL_DATADIR}/libxml2/cmake
 )
 
 install(
 	FILES ${CMAKE_CURRENT_BINARY_DIR}/libxml2-config.cmake
-	DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libxml2-${PROJECT_VERSION}
+	DESTINATION ${CMAKE_INSTALL_DATADIR}/libxml2/cmake
 	COMPONENT development
 )
 
@@ -601,13 +601,13 @@ write_basic_package_version_file(
 
 install(
 	FILES ${CMAKE_CURRENT_BINARY_DIR}/libxml2-config-version.cmake
-	DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libxml2-${PROJECT_VERSION}
+	DESTINATION ${CMAKE_INSTALL_DATADIR}/libxml2/cmake
 	COMPONENT development
 )
 
 install(
 	EXPORT LibXml2
-	DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libxml2-${PROJECT_VERSION}
+	DESTINATION ${CMAKE_INSTALL_DATADIR}/libxml2/cmake
 	NAMESPACE LibXml2::
 	FILE libxml2-export.cmake
 	COMPONENT development
@@ -649,10 +649,10 @@ if(WIN32)
 	set(prefix "\$(cd \"\$(dirname \"\$0\")\"; pwd -P)/..")
 endif()
 configure_file(xml2-config.in xml2-config @ONLY)
-install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/xml2-config DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT development)
+install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/xml2-config DESTINATION ${CMAKE_INSTALL_DATADIR}/libxml2 COMPONENT development)
 
 if(UNIX)
-	install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/libxml.m4 DESTINATION ${CMAKE_INSTALL_DATADIR}/aclocal)
+	install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/libxml.m4 DESTINATION ${CMAKE_INSTALL_DATADIR}/libxml2/aclocal)
 endif()
 
 set(XML_INCLUDEDIR "-I${CMAKE_INSTALL_FULL_INCLUDEDIR}/libxml2")
-- 
2.39.2.windows.1

