From 1cc67f3517de892fe87753da4fbff0e8ec86b8f8 Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Tue, 23 Aug 2022 14:50:24 -0700
Subject: [PATCH 2/2] Remove library suffix on Windows

---
 CMakeLists.txt | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4c6b2c42..61f13290 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -454,25 +454,6 @@ set_target_properties(
         SOVERSION ${LIBXML_MAJOR_VERSION}
 )
 
-if(MSVC)
-	if(BUILD_SHARED_LIBS)
-		set_target_properties(
-			LibXml2
-			PROPERTIES
-			DEBUG_POSTFIX d
-		)
-	else()
-		set_target_properties(
-			LibXml2
-			PROPERTIES
-			DEBUG_POSTFIX sd
-			MINSIZEREL_POSTFIX s
-			RELEASE_POSTFIX s
-			RELWITHDEBINFO_POSTFIX s
-		)
-	endif()
-endif()
-
 install(FILES ${LIBXML2_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libxml2/libxml COMPONENT development)
 
 install(
-- 
2.43.0.windows.1

