From c63153119b4d3c2d6a9d6f01d0b2539f73a4746e Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Mon, 27 Jun 2022 18:34:05 -0700
Subject: [PATCH 2/2] Add a shared ngtcp2_crypto_boringssl target

---
 crypto/boringssl/CMakeLists.txt | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/crypto/boringssl/CMakeLists.txt b/crypto/boringssl/CMakeLists.txt
index 60d2d414..f955be51 100644
--- a/crypto/boringssl/CMakeLists.txt
+++ b/crypto/boringssl/CMakeLists.txt
@@ -43,6 +43,26 @@ foreach(name libngtcp2_crypto_boringssl.pc)
   configure_file("${name}.in" "${name}" @ONLY)
 endforeach()
 
+# Public shared library
+if(ENABLE_SHARED_LIB)
+  add_library(ngtcp2_crypto_boringssl SHARED ${ngtcp2_crypto_boringssl_SOURCES})
+  set_target_properties(ngtcp2_crypto_boringssl PROPERTIES
+    COMPILE_FLAGS "${WARNCFLAGS}"
+    VERSION ${LT_VERSION}
+    SOVERSION ${LT_VERSION}
+    C_VISIBILITY_PRESET hidden
+    POSITION_INDEPENDENT_CODE ON
+  )
+  target_include_directories(ngtcp2_crypto_boringssl PUBLIC
+    ${ngtcp2_crypto_boringssl_INCLUDE_DIRS})
+  target_link_libraries(ngtcp2_crypto_boringssl ngtcp2 ${BORINGSSL_LIBRARIES})
+
+  install(TARGETS ngtcp2_crypto_boringssl
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
+endif()
+
 if(ENABLE_STATIC_LIB)
   # Public static library
   add_library(ngtcp2_crypto_boringssl_static STATIC ${ngtcp2_crypto_boringssl_SOURCES})
-- 
2.41.0.windows.1

